### PR TITLE
Add first tests for monitor_t and fix monitor_t::abort

### PR DESCRIPTION
**Pull request by sigiesec**

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -23,11 +23,13 @@ add_executable(
...
     active_poller.cpp
     multipart.cpp
+	monitor.cpp
     )
 
 target_link_libraries(
     unit_tests
-    PRIVATE gtest_main
+	PRIVATE gtest
+	PRIVATE gmock_main
```
Comment (kurdybacha) [inline tests/CMakeLists.txt]: is the indentation intentional ?

File: tests/monitor.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+TEST(monitor, init_check)
+{
+    zmq::context_t ctx;
+    zmq::socket_t bind_socket(ctx, ZMQ_DEALER);
+
+    bind_socket.bind("tcp://127.0.0.1:*");
+    char endpoint[255];
+    size_t endpoint_len = sizeof(endpoint);
+    bind_socket.getsockopt(ZMQ_LAST_ENDPOINT, &endpoint, &endpoint_len);
```
Comment (kurdybacha) [inline tests/monitor.cpp]: you have this logic for endpoint, two sockets in testutil.hpp, could you try to reuse, please?

File: tests/monitor.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#ifdef ZMQ_CPP11
+TEST(monitor, init_abort)
+{
+    zmq::context_t ctx;
+    zmq::socket_t bind_socket(ctx, zmq::socket_type::dealer);
+
+    bind_socket.bind("tcp://127.0.0.1:*");
+    char endpoint[255];
+    size_t endpoint_len = sizeof(endpoint);
+    bind_socket.getsockopt(ZMQ_LAST_ENDPOINT, &endpoint, &endpoint_len);
```
Comment (kurdybacha) [inline tests/monitor.cpp]: same here

File: tests/monitor.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    EXPECT_CALL(monitor, on_event_connected(testing::_, testing::_))
+      .Times(testing::AtLeast(1));
+
+    auto thread = std::thread([&monitor] {
+        while (monitor.check_event(-1)) {
+        }
+    });
+
+    connect_socket.connect(endpoint);
+    std::this_thread::sleep_for(std::chrono::milliseconds(250));
```
Comment (kurdybacha) [inline tests/monitor.cpp]: that made total running time of unit_tests ~13 times slower (~21ms -> ~270ms).  Why do you need this? You want to abort while events are pending?

File: tests/CMakeLists.txt (hunk 1 above)
Comment (sigiesec) [reply tests/CMakeLists.txt]: No... probably this is a problem with tabs vs. spaces

**Pull request merged**